### ISSUE TITLE
Pulumi package version incompatible with pulumi/actions

### ISSUE THREAD

**Issue by bombillazo** [pulumi/pulumi-aws#1928]
### What happened?

After updating our Pulumi packages to the latest versions:
```
    "@pulumi/aws": "^5.3.0",
    "@pulumi/awsx": "^0.40.0",
    "@pulumi/pulumi": "^3.30.0",
    "@pulumi/random": "^4.5.0",
```

Our Github Actions pulumi workflow fails witht eh following error:

```
aws:acm:Certificate (api-cert):
    error: 1 error occurred:
    	* error configuring Terraform AWS Provider: failed to get shared config profile, profile
```

Our only solution is to use older versions:
```
    "@pulumi/aws": "^4.38.1",
    "@pulumi/awsx": "^0.33.0",
    "@pulumi/pulumi": "^3.26.1",
    "@pulumi/random": "^4.4.2",
```

### Steps to reproduce

Install the latest Pulumi packages:
```
    "@pulumi/aws": "^5.3.0",
    "@pulumi/awsx": "^0.40.0",
    "@pulumi/pulumi": "^3.30.0",
    "@pulumi/random": "^4.5.0",
```

Setup a basic github actions workflow to preview 

```
      - name: Pulumi preview
        uses: pulumi/actions@v3
        with:
          command: preview
          stack-name: ${{ env.PULUMI_STACK }}
          work-dir: infra
        env:
          PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
          AWS_ACCESS_KEY_ID: ${{ env.AWS_ACCESS_KEY_ID }}
          AWS_SECRET_ACCESS_KEY: ${{ env.AWS_SECRET_ACCESS_KEY }}
          AWS_REGION: ${{ env.AWS_REGION }}
```

### Expected Behavior

Preview command runs successfully.

### Actual Behavior

Preview fails at random points of the rpeview with this error:

```
aws:acm:Certificate (api-cert):
    error: 1 error occurred:
    	* error configuring Terraform AWS Provider: failed to get shared config profile, profile
```

### Versions used

    "@pulumi/aws": "^5.3.0",
    "@pulumi/awsx": "^0.40.0",
    "@pulumi/pulumi": "^3.30.0",
    "@pulumi/random": "^4.5.0",

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by guineveresaenger** [comment]
Hi! Thank you for filing this issue.

This is due to a [change in the order of override for env vars Terraform uses](https://github.com/pulumi/pulumi-aws/issues/1897#issuecomment-1106899341).

tl;dr: You'll want to set your AWS_PROFILE in your pulumi config in your actions.
